File: Assets/Script/Channel/BaseChannel.cs
namespace Common.GameChannel
{
    public abstract class BaseChannel
    {
        //基础
        public abstract void Init();

        public abstract void DownloadGame(params object[] paramList);
        /// <summary>
        /// 安装apk
        /// 这个地方需要调用原生代码去安装，目前来说
        /// 不去做，有能力的可以去做 安卓 ios的原生代码
        /// </summary>
        public abstract void InstallApk();
        public abstract string GetProductName();

        public virtual bool IsInternalChannel()
        {
            return false;
        }
    }
}

File: Assets/.lpproj/Apis/Unity/CS/Common/GameChannel/BaseChannel.lua
---@module CS.Common.GameChannel
CS.Common.GameChannel = {}

---@class CS.Common.GameChannel.BaseChannel : CS.System.Object
CS.Common.GameChannel.BaseChannel = {}

function CS.Common.GameChannel.BaseChannel:Init()
end

---@param paramList : CS.System.Object[]
function CS.Common.GameChannel.BaseChannel:DownloadGame(paramList)
end

---@return CS.System.Boolean
function CS.Common.GameChannel.BaseChannel:IsGooglePlay()
end

---@return CS.System.String
function CS.Common.GameChannel.BaseChannel:GetBundleID()
end

function CS.Common.GameChannel.BaseChannel:InstallApk()
end

---@return CS.System.String
function CS.Common.GameChannel.BaseChannel:GetProductName()
end

function CS.Common.GameChannel.BaseChannel:DataTrackInit()
end

---@return CS.System.String
function CS.Common.GameChannel.BaseChannel:GetCompanyName()
end

---@return CS.System.Boolean
function CS.Common.GameChannel.BaseChannel:IsInternalChannel()
end

function CS.Common.GameChannel.BaseChannel:Login()
end

function CS.Common.GameChannel.BaseChannel:Logout()
end

---@param paramList : CS.System.Object[]
function CS.Common.GameChannel.BaseChannel:Pay(paramList)
end